index.js

```diff
@@ -303,6 +343,16 @@ function _handleWrapError(instance, error, name, context, args, options, callbac
   }
 }
 
+/**
+ *
```

Comment:
i dont quite know what this function does, so i left it empty

Reply:
I realize this function is a bit tricky to read and the name could be better.

This function executes pre hooks, then calls the wrapped function, then executes post hooks. So this function or `createWrapper()` (which just calls this function) is how we recommend executing hooks, unless you want more fine-grained control and want to call `execPre()` and `execPost()` yourself.